test/unit/test_keyring_raw_rsa.py

```diff
@@ -0,0 +1,311 @@
+# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+"""Unit tests for Raw AES keyring."""
+
+import pytest
+from cryptography.hazmat.primitives.asymmetric import rsa
+from pytest_mock import mocker  # noqa pylint: disable=unused-import
+
+import aws_encryption_sdk.key_providers.raw
+import aws_encryption_sdk.keyring.raw_keyring
+from aws_encryption_sdk.identifiers import KeyringTraceFlag, WrappingAlgorithm
+from aws_encryption_sdk.internal.crypto.wrapping_keys import WrappingKey
+from aws_encryption_sdk.keyring.base import Keyring
+from aws_encryption_sdk.keyring.raw_keyring import RawRSAKeyring
+
+from .test_utils import (
+    _BACKEND,
+    _DATA_KEY,
+    _ENCRYPTED_DATA_KEY_RSA,
+    _ENCRYPTION_CONTEXT,
+    _KEY_ID,
+    _KEY_SIZE,
+    _PROVIDER_ID,
+    _PUBLIC_EXPONENT,
+    get_decryption_materials_with_data_encryption_key,
+    get_decryption_materials_without_data_encryption_key,
+    get_encryption_materials_with_data_encryption_key,
+    get_encryption_materials_without_data_encryption_key,
+)
+
+try:  # Python 3.5.0 and 3.5.1 have incompatible typing modules
+    from typing import Iterable  # noqa pylint: disable=unused-import
+except ImportError:  # pragma: no cover
+    # We only actually need these imports when running the mypy checks
+    pass
+
+pytestmark = [pytest.mark.unit, pytest.mark.local]
+
+
+@pytest.fixture
+def patch_generate_data_key(mocker):
+    mocker.patch.object(aws_encryption_sdk.keyring.raw_keyring, "generate_data_key")
+    return aws_encryption_sdk.keyring.raw_keyring.generate_data_key
+
+
+@pytest.fixture
+def patch_decrypt_data_key(mocker):
```

Comment:
Ditto on naming: make it clear if I'm just looking at a test where this fixture is being consumed, what this is patching.